plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/commands/CodeTransformActionMessage.kt

```diff
@@ -0,0 +1,14 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codemodernizer.commands
+
+import software.aws.toolkits.jetbrains.services.amazonq.messages.AmazonQMessage
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.CodeModernizerJobCompletedResult
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.MavenCopyCommandsResult
+
+data class CodeTransformActionMessage(
+    val command: CodeTransformCommand,
+    val mavenBuildResult: MavenCopyCommandsResult? = null,
+    val transformResult: CodeModernizerJobCompletedResult? = null
+) : AmazonQMessage
```

Comment:
nit: May be better to have subclasses like MavenBuildResultActionMessage or TransformResultActionMessage.